extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/BeanArchiveProcessor.java

```diff
@@ -157,6 +165,33 @@ private boolean isAdditionalBeanArchive(ApplicationArchive archive,
         return false;
     }
 
+    private boolean possiblyBeanArchive(ApplicationArchive archive) {
+        return archive.apply(tree -> {
+            boolean result = true;
+            for (String beansXml : List.of("META-INF/beans.xml", "WEB-INF/beans.xml")) {
```

Comment:
I don't think you can use WAR dependencies in quarkus but let's make it safe  ;-)

Reply:
I agree, but the same set of files is inspected by the `isExplicitBeanArchive` method. I would be in favor of only looking at `META-INF/beans.xml`.

Reply:
Do we make this change in this pr?

Reply:
I suggest not, but I don't really have a good reason other than it seems out of the scope (of improving spec compatibility). I can add a commit to get rid of `WEB-INF/beans.xml` usage if you want.

Reply:
Let's address this in a separate pull request..